sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/repository/alternative/SqlEntityRepository.java

```diff
@@ -0,0 +1,166 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED                          
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;                               
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER                              
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.sodalite.server.repository.alternative;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.icgc.dcc.sodalite.server.utils.JsonUtils;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.Update;
+import org.skife.jdbi.v2.util.StringMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import lombok.SneakyThrows;
+import lombok.val;
+
+class SqlEntityRepository extends EntityRepository {
+
+  @Autowired
+  Handle h;
+  boolean parentsExist;
+
+  Update create;
+  Update update;
+  Update delete;
+
+  List<String> fields;
+  private String tableName;
+  private String parentIdField;
+
+  SqlEntityRepository(Handle h, String tableName, String parentIdField, List<String> fields) {
+    this.h = h;
+    this.tableName = tableName;
+    this.parentIdField = parentIdField;
+    this.fields = fields;
+    val insert = sqlCmdCreate(tableName, parentIdField, fields);
+    create = h.createStatement(insert);
+  }
+
+  private boolean parentsExist() {
+    return parentIdField == null;
+  }
+
+  String sqlCmdCreate(String tableName, String parentIdField, List<String> fields) {
+    String insert = "INSERT INTO " + tableName + " (id, studyId, ";
+    if (parentsExist()) {
+      insert += parentIdField;
+    }
+    insert += fieldList() + ") VALUES (?, ?";
+    if (parentsExist) {
+      insert += " ,?";
+    }
+    insert += bindings() + ")";
+    return insert;
+  }
+
+  @Override
+  public void create(String id, String parentId, String studyId, JsonNode data) {
+    int i = 0;
+    this.create.bind(i++, id);
+    this.create.bind(i++, studyId);
+
+    if (parentsExist) {
+      this.create.bind(i++, parentId);
+    }
+
+    for (val f : fields) {
+      this.create.bind(i++, data.get(f).asText());
+    }
+
+    create.execute();
+
+  }
+
+  private String bindings() {
+    String s = "";
+    for (int i = 1; i <= fields.size(); i++) {
+      s += " ,?";
+    }
+    return s;
+  }
+
+  private String fieldList() {
+    return fields.stream().collect(Collectors.joining(","));
+  }
+
+  @SneakyThrows
+  @Override
+  public JsonNode read(String id) {
+    val rs = h.select("select * from " + tableName + " WHERE ID = " + id);
+    val row = rs.get(0);
+
+    String json = "{";
```

Comment:
You should create the JsonNode directly rather than mutating a string reference several times and then parsing it.

```java
  /**
   * Constants
   */
   private static final ObjectMapper MAPPER = new ObjectMapper();
.
.
.
  @SneakyThrows
  @Override
  public JsonNode read(String id) {
    val rs = handler.select(format("SELECT * from %s WHERE ID = %s", tableName, id));
    val row = rs.get(0);


    val json = MAPPER.createObjectNode();
```